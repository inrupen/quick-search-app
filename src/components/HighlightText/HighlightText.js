import React from 'react'
import styled from 'styled-components'

const Div = styled.div`
    color: #6d6d6d;
    font-size: 15px;
    padding: 0.4rem;
    display: block;
    text-align: left;
    border-right: 1.3px solid #e2e2e2;
    border-left: 1.3px solid #e2e2e2;
    border-bottom: 1.3px solid #e2e2e2;

    strong {
        color: #000000;
        font-weight: 700;
    }
`

export default function Highlight({ searchWord, string, number }) {
    function strongString(input) {
        return `<strong class="strong">${input}</strong>`
    }
    function highlightString() {
        let finalString = ""
        if (string && number) {
            if (searchWord && string.includes(searchWord)) {
                const searchString = string
                const searchThisInString = searchWord
                const splitString = searchString.split(searchThisInString)
                if (searchString.startsWith(searchThisInString)) {
                    finalString =
                        searchThisInString + strongString(splitString[1])
                } else if (searchString.endsWith(searchThisInString)) {
                    finalString =
                        strongString(splitString[0]) + searchThisInString
                } else {
                    finalString =
                        strongString(splitString[0]) +
                        searchThisInString +
                        strongString(splitString[1])
                }
            } else {
                finalString = `<strong class="strong">${string}</strong>`
            }
            finalString += ` (${number})`
        }
        return finalString
    }

    return <Div dangerouslySetInnerHTML={{ __html: highlightString() }} />
}
