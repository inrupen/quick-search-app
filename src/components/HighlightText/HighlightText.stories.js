import React from "react"
import HighlightText from "./HighlightText"

export default {
    component: HighlightText,
    title: "Components/HighlightText",
}

const mockData = require("../../../api/_search.get.json")
const suggestions = mockData.suggestions
const data = suggestions[0]

export const withoutText = () => <HighlightText />
export const withText = () => (
    <HighlightText
        searchWord={"truie"}
        string={data.searchterm}
        number={data.nrResults}
    />
)
