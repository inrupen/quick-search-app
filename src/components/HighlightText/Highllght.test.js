import React from "react"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import HighlightText from "./HighlightText"

const mockData = require("../../../api/_search.get.json")
const suggestions = mockData.suggestions
const data = suggestions[0]

let HighlightWithoutText
let HighlightWithText

beforeEach(() => {
    HighlightWithoutText = render(<HighlightText />)
    HighlightWithText = render(
        <HighlightText
            searchWord={"truie"}
            string={data.searchterm}
            number={data.nrResults}
        />
    )
})

afterEach(() => {
    HighlightWithoutText.unmount()
    HighlightWithText.unmount()
})

describe("highlight text component", () => {
    it("renders no highlighted text-when no props", () => {
        const { container  } = render( <HighlightText /> )
        expect(HighlightWithoutText).toMatchSnapshot()
        expect(container).not.toContainHTML("strong")

    })

    it("renders highlighted text-with props", () => {
        const { container } = render(
            <HighlightText
                searchWord={"truie"}
                string={data.searchterm}
                number={data.nrResults}
            />
        )
        expect(HighlightWithText).toMatchSnapshot()
        expect(container).toContainHTML("strong")
    })
})
