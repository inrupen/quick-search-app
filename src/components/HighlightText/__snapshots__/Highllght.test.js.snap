// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlight text component renders highlighted text-with props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBQB jlKTlV"
      />
    </div>
    <div>
      <div
        class="sc-bdfBQB jlKTlV"
      >
        <strong
          class="strong"
        >
          heren 
        </strong>
        truie
        <strong
          class="strong"
        >
          n
        </strong>
         (1100)
      </div>
    </div>
    <div>
      <div
        class="sc-bdfBQB jlKTlV"
      >
        <strong
          class="strong"
        >
          heren 
        </strong>
        truie
        <strong
          class="strong"
        >
          n
        </strong>
         (1100)
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBQB jlKTlV"
    >
      <strong
        class="strong"
      >
        heren 
      </strong>
      truie
      <strong
        class="strong"
      >
        n
      </strong>
       (1100)
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`highlight text component renders no highlighted text-when no props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBQB jlKTlV"
      />
    </div>
    <div>
      <div
        class="sc-bdfBQB jlKTlV"
      >
        <strong
          class="strong"
        >
          heren 
        </strong>
        truie
        <strong
          class="strong"
        >
          n
        </strong>
         (1100)
      </div>
    </div>
    <div>
      <div
        class="sc-bdfBQB jlKTlV"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBQB jlKTlV"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
